cds/webapi/C#/BlazorCDS/BlazorCDS/Pages/FetchAccounts.razor

```diff
@@ -0,0 +1,228 @@
+﻿@page "/fetchaccounts"
+@using Microsoft.AspNetCore.Components.WebAssembly.Authentication
+@using System.Net.Http.Headers
+@using System.Net.Http.Json
+@using Microsoft.Extensions.Logging
+@using System.Text.Json.Serialization
+@inject IAccessTokenProvider TokenProvider
+@inject IHttpClientFactory ClientFactory
+@inject ILogger<FetchAccounts> logger
+@inject Models.AppState AppState
+@implements IDisposable
+
+<AuthorizeView>
+    @*Only show the list if the user is signed in and authorized*@
+    <Authorized>
+        <h3>Fetch Accounts</h3>
+        <br /><label>SelectedEnvUrl: @AppState.SelectedEnvUrl</label>
+
+        @if (!string.IsNullOrEmpty(AppState.SelectedEnvUrl))
+        {
+            if (accounts != null)
+            {
+                <table class="table">
+                    <thead>
+                        <tr>
+                            <th>Name</th>
+                            <th>Main Phone</th>
+                            <th>City</th>
+                            <th>Primary Contact</th>
+                            <th>Email (Primary Contact)</th>
+                        </tr>
+                    </thead>
+                    <tbody>
+                        @foreach (Account account in accounts.value)
+                        {
+                            <tr id="@account.accountid">
+                                <td>
+                                    @((account.name != null)
+                                ? account.name
+                                : string.Empty)
+                                </td>
+                                <td>
+                                    @((account.telephone1 != null)
+                                ? account.telephone1
+                                : string.Empty)
+                                </td>
+                                <td>
+                                    @((account.address1_city != null)
+                                ? account.address1_city
+                                : string.Empty)
+                                </td>
+                                <td>
+                                    @((account.primarycontactid != null)
+                                ? (account.primarycontactid.fullname != null
+                                    ? account.primarycontactid.fullname
+                                    : string.Empty)
+                                : string.Empty)
+                                </td>
+                                <td>
+                                    @((account.primarycontactid != null)
```

Comment:
```
                            <td>
                                @((account.primarycontactid != null)
                                ? (account.primarycontactid.emailaddress1 !=null
                                    ? account.primarycontactid.emailaddress1
                                    : string.Empty)
                                : string.Empty)
                            </td>
```

This is the same code I used here: [Tutorial: Create an ASP.NET Core Blazor WebAssembly App using Common Data Service > Step 4: Apply code changes > Add \Pages\FetchAccounts.razor](https://docs.microsoft.com/en-us/powerapps/developer/common-data-service/walkthrough-blazor-webassembly-single-tenant#add-pagesfetchaccountsrazor)

I suspect that there is a more elegant way to manage this. 
Not required to approve this PR, but if anyone has any ideas I'd love to hear them.
Perhaps using [?? and ??= operators (C# reference)](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/null-coalescing-operator)